metaflow/plugins/kfp/tests/.gitlab-ci.yml

```diff
@@ -1,211 +1,142 @@
-  artifacts:
-    paths:
-      - ${_IMAGE_TAG_FILE} # we pass this file as an artifact to be read in the testing stages
 
-test:internal:
-  extends: 
-    - .devex_internal_eks # sets up AWS environment variables
-    - .generate_kubeconfig # creates the .kube directory for the appropriate cluster
+.version: &version |
+  METAFLOW_VERSION=$(cat $METAFLOW_VERSION_PATH | sed -nr "s/^version = ['\"]([^'\"]*)['\"]$/\1/p")
+
+  if [ "${CI_COMMIT_BRANCH}" = "${CI_DEFAULT_BRANCH}" ]; then
+    PY_LIBRARY_VERSION="${METAFLOW_VERSION}.${CI_PIPELINE_IID}"
+    IMAGE_TAG="${METAFLOW_VERSION}.${CI_PIPELINE_IID}"
+  else
+    PY_LIBRARY_VERSION="0.0.${CI_PIPELINE_IID}+${CI_COMMIT_SHORT_SHA}"
+    IMAGE_TAG="0.0.${CI_PIPELINE_IID}"
+  fi
+
+.test:
+  extends: .generate_kubeconfig
+  stage: test
   variables:
     GIT_STRATEGY: none
     PIPELINE_VERSION: "1.0"
-  stage: test
   script:
-    - *path-to-host-file-directory
-    - export BUILT_IMAGE_FULL_PATH=$( cat ${_IMAGE_TAG_FILE} ) # reading the path to the Docker image on Artifactory
+    - *path-to-host-file-directory  
+    - export BUILT_IMAGE_FULL_PATH=$( cat ${IMAGE_REPOSITORY_TAG_PATH} )
     - cp -r /root/.kube .
     - docker run
         --rm
         -v ${SHARED_PATH}/.kube:/home/zservice/.kube
-        -v ${SHARED_PATH}/public:/home/zservice/public 
+        -v ${SHARED_PATH}/public:/home/zservice/public  
         -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID 
         -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY 
         -e AWS_SESSION_TOKEN=$AWS_SESSION_TOKEN 
         -e AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION 
+        -e KFP_RUN_URL_PREFIX=$KFP_RUN_URL_PREFIX 
+        -e KFP_SDK_NAMESPACE=$KFP_SDK_NAMESPACE 
+        -e USER=$GITLAB_USER_EMAIL
         ${BUILT_IMAGE_FULL_PATH}
         bash -c "
-          export KFP_RUN_URL_PREFIX=https://kubeflow.corp.dev-k8s.zg-aip.net/ && 
-          export KFP_SDK_NAMESPACE=metaflow-integration-testing-internal && 
-          export USER=${GITLAB_USER_EMAIL} &&
           cd /home/zservice/metaflow/metaflow/plugins/kfp/tests &&
           python -m pytest -s -n 3 run_integration_tests.py --image ${BUILT_IMAGE_FULL_PATH} --opsgenie-api-token ${OPSGENIE_API_TOKEN} --cov-config=setup.cfg
         "
   artifacts:
     when: always
     paths:
       - public
+  
+
+build:docker:
+  stage: build
+  script:
+    - *version
+    - export IMAGE_REPOSITORY_TAG=${DOCKER_REPO_URL}/${CI_PROJECT_NAMESPACE}/${PY_LIBRARY_NAME}:${IMAGE_TAG}
+    - 'docker build --no-cache -f metaflow/plugins/kfp/tests/Dockerfile -t ${IMAGE_REPOSITORY_TAG} .'
+    - 'echo ${DOCKER_API_KEY} | docker login -u ${DOCKER_USERNAME} --password-stdin ${DOCKER_REPO_URL}'
+    - STDERR=$(docker push ${IMAGE_REPOSITORY_TAG} 2>&1 >/dev/null) || true
+    - echo $STDERR
+    - |
+      if [ -z "$STDERR" ] || [[ $STDERR == *"manifest invalid"* ]]; then
+        echo "Successful!"
+      else
+        echo "Error in pushing image to Artifactory."
```

Comment:
Can we echo the `$STDERR` so it's in the error logs?

Reply:
Sure fixed!